File: fix_nav.py
import os
import re

TARGET_DIR = r"E:\TENET-5.github.io"

# The standard nav block for inner pages
STANDARD_NAV = """    <nav aria-label="Primary">
      <a href="index.html">Home</a>
      <a href="news.html">News</a>
      <a href="daily-briefing.html">Briefing</a>
      <a href="investigations.html">Investigations</a>
      <a href="argument.html">The Case</a>
      <a href="evidence-index.html">Evidence</a>
      <a href="information-architecture.html">Map</a>
    </nav>"""

nav_regex = re.compile(r'^[ \t]*<nav[^>]*aria-label="Primary"[^>]*>.*?</nav>', re.DOTALL | re.MULTILINE)

def process_file(filepath):
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()
    
    if 'aria-label="Primary"' in content:
        # Don't touch index.html cover-nav as it has class="cover-nav"
        if "index.html" in os.path.basename(filepath):
            return
            
        new_content, count = nav_regex.subn(STANDARD_NAV, content)
        if count > 0 and new_content != content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated {filepath}")

for root, _, files in os.walk(TARGET_DIR):
    # skip node_modules
    if "node_modules" in root:
        continue
    for file in files:
        if file.endswith('.html'):
            process_file(os.path.join(root, file))

File: fix_nav.js
const fs = require('fs');
const path = require('path');

const directoryPath = path.join(__dirname);
const files = fs.readdirSync(directoryPath).filter(file => file.endsWith('.html'));

const links = [
    { href: 'index.html', text: 'Home' },
    { href: 'news.html', text: 'News' },
    { href: 'daily-briefing.html', text: 'Briefing' },
    { href: 'investigations.html', text: 'Investigations' },
    { href: 'argument.html', text: 'The Case' },
    { href: 'evidence-index.html', text: 'Evidence' },
    { href: 'about.html', text: 'About' }
];

let updatedCount = 0;

files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    
    // Regular expression to match <nav aria-label="Primary">...</nav>
    // or <nav class="cover-nav" aria-label="Primary">...</nav>
    const navRegex = /(<nav\s+[^>]*aria-label="Primary"[^>]*>)([\s\S]*?)(<\/nav>)/i;
    
    if (navRegex.test(content)) {
        // Build the new inner HTML
        let newInnerHtml = '\n';
        links.forEach(link => {
            const isCurrent = (file === link.href) ? ' aria-current="page"' : '';
            newInnerHtml += `      <a href="${link.href}"${isCurrent}>${link.text}</a>\n`;
        });
        newInnerHtml += '    ';
        
        // Replace the content
        content = content.replace(navRegex, (match, openTag, oldInner, closeTag) => {
            return `${openTag}${newInnerHtml}${closeTag}`;
        });
        
        fs.writeFileSync(file, content, 'utf8');
        updatedCount++;
    }
});

console.log(`Successfully unified the top navigation bar across ${updatedCount} HTML files!`);
